include/index.h

```diff
@@ -377,6 +384,7 @@ template <typename T, typename TagT = uint32_t, typename LabelT = uint32_t> clas
     std::string _labels_file;
     std::unordered_map<LabelT, uint32_t> _label_to_medoid_id;
     std::unordered_map<uint32_t, uint32_t> _medoid_counts;
+    std::unordered_map<LabelT, uint32_t> _label_counts;
```

Comment:
do we need to protect any of the label related data structures with a lock? If so, can we document it in index.h?

Reply:
This i believe was added by David, I don't think its being used as such. I will remove it.